compiler/rustc_trait_selection/src/solve/inspect.rs

```diff
@@ -145,11 +147,15 @@ impl<'tcx> From<WipGoalCandidate<'tcx>> for DebugSolver<'tcx> {
 
 pub struct ProofTreeBuilder<'tcx> {
     state: Option<Box<DebugSolver<'tcx>>>,
+    disable_global_cache: DisableGlobalCache,
 }
```

Comment:
vibe, can we go with
```rust
struct ProofTreeBuilder<'tcx> {
    data: Option<Box<BuilderData<'tcx>>,
}

struct BuilderData<'tcx> {
    state: DebugSolver<'tcx>,
    disable_global_cache: DisableGlobalCache,
}
```
I am unhappy that even with a noop proof tree ther's still a `disable_global_cache` you can access.

Reply:
or hmm, i guess it's fine